alacritty_terminal/src/term/mod.rs

```diff
@@ -1132,49 +1034,31 @@ impl<T> Term<T> {
     pub fn resize(&mut self, size: &SizeInfo) {
         let old_cols = self.grid.num_cols();
         let old_lines = self.grid.num_lines();
-        let mut num_cols = size.cols();
-        let mut num_lines = size.lines();
+        let num_cols = max(size.cols(), Column(2));
+        let num_lines = max(size.lines(), Line(2));
 
         if old_cols == num_cols && old_lines == num_lines {
             debug!("Term::resize dimensions unchanged");
             return;
         }
 
-        self.grid.selection = None;
-        self.alt_grid.selection = None;
-
-        // Should not allow less than 2 cols, causes all sorts of checks to be required.
-        if num_cols <= Column(1) {
-            num_cols = Column(2);
-        }
-
-        // Should not allow less than 2 lines, causes all sorts of checks to be required.
-        if num_lines <= Line(1) {
-            num_lines = Line(2);
-        }
-
         debug!("New num_cols is {} and num_lines is {}", num_cols, num_lines);
 
         let is_alt = self.mode.contains(TermMode::ALT_SCREEN);
-        let alt_cursor_point =
-            if is_alt { &mut self.cursor_save.point } else { &mut self.cursor_save_alt.point };
 
         // Resize grids to new size
-        self.grid.resize(!is_alt, num_lines, num_cols, &mut self.cursor.point, &Cell::default());
-        self.alt_grid.resize(is_alt, num_lines, num_cols, alt_cursor_point, &Cell::default());
+        let mut vi_cursor_absolute = self.visible_to_buffer(self.vi_mode_cursor.point);
+        let mut resize: [&mut dyn Resize; 2] = [&mut vi_cursor_absolute, &mut self.selection];
```

Comment:
It seems like vi cursor is failing to reflow properly from time to time. It works fine for horizontal resize, but when I do diagonal resize it's failing and changes line and column.

Reply:
Ok, it's due to clamping of cursor to viewport, I'm not sure that it's a bug of expected behavior.

Reply:
It's expected behavior. At least if you're talking about what I think you're talking about.

Seems like the best approach to just clamp to within the viewport.